### PR TITLE
Add service attachment roncile connections

### PR DESCRIPTION
**Pull request by diogoEsteves42**
Solves the issue: https://github.com/hashicorp/terraform-provider-google/issues/14915

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added `reconcile_connections` for `google_compute_service_attachment`.
```

### CODE REVIEW

**Comment by c2thorn** [comment]
Please @ me when this is ready to be reviewed and run in our CI.

**Comment by diogoEsteves42** [comment]
Hi @c2thorn thanks for the message. I believe this PR will remain in draft mode for a couple of days, I've just talked with people from google to figure out why my tests just don't work, and they told me that the field was not released yet.

I'll ping you when I open the PR to follow with the review
Thanks

**Comment by diogoEsteves42** [comment]
Hi @c2thorn, the field was released and I'm able to test it.
PR ready to review, thanks.

**Comment by melinath** [comment]
drive by re-run - tgc failures should be resolved now. /gcbrun

File: mmv1/templates/terraform/examples/service_attachment_reconcile_connections.tf.erb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  domain_names             = ["gcp.tfacc.hashicorptest.com."]
+  enable_proxy_protocol    = true
+  connection_preference    = "ACCEPT_MANUAL"
+  nat_subnets              = [google_compute_subnetwork.psc_ilb_nat.id]
+  target_service           = google_compute_forwarding_rule.psc_ilb_target_service.id
+
+  consumer_reject_lists = ["673497134629", "482878270665"]
+
+  consumer_accept_lists {
+    project_id_or_num = "658859330310"
```
Comment (c2thorn) [inline mmv1/templates/terraform/examples/service_attachment_reconcile_connections.tf.erb:15]: What project number was used for this?

Comment (diogoEsteves42) [reply mmv1/templates/terraform/examples/service_attachment_reconcile_connections.tf.erb:15]: Hi @c2thorn, for this values I just follow the example of this test https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/examples/service_attachment_explicit_projects.tf.erb

Comment (c2thorn) [reply mmv1/templates/terraform/examples/service_attachment_reconcile_connections.tf.erb:15]: ahh, I see. thanks.

**Review by c2thorn — APPROVED** [review]

**Comment by diogoEsteves42** [comment]
Hi @c2thorn, I believe you can help me with something.
We have some customers complaining about this resource that when they update, Terraform deletes and then creates a completely new resource, instead of just updating the existing one. I believe this is because they are updating the immutable fields.
Is that correct?

**Comment by c2thorn** [comment]
> Hi @c2thorn, I believe you can help me with something. We have some customers complaining about this resource that when they update, Terraform deletes and then creates a completely new resource, instead of just updating the existing one. I believe this is because they are updating the immutable fields. Is that correct?

yes that is likely the case. Terraform should be displaying what it will do and why during `terraform apply` or `terraform plan`. If they are auto-accepting they likely aren't seeing it. If Terraform was destroying and recreating outside of updating an immutable field, it would be a very serious bug

**Comment by c2thorn** [comment]
I've been trying to figure out why the `terraform-google-conversion` tests are failing, and I think it is due to an unrelated change that has since been fixed upstream. Do you mind rebasing onto the latest changes on `main` so I can rerun and verify? @diogoEsteves42

**Comment by diogoEsteves42** [comment]
@c2thorn, thanks for the explanation.
I believe it's not a case of a bug

Done the update!

**Review by c2thorn — APPROVED** [review]

**Pull request merged**